### PR TITLE
Completed DP 1

### PR DESCRIPTION
**Pull request by goodguygoddy**

### CODE REVIEW

File: Problem_1.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Time Complexity : O(n*amount)
+ * Space Complexity : O(1)
```
Comment (ishika2741) [inline Problem_1.cpp]: Check space complexity

File: Problem_2.cpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
+/*
+ * Time Complexity : O(n)
+ * Space Complexity : O(1)
```
Comment (ishika2741) [inline Problem_2.cpp]: Check space complexity

**Review by ishika2741 — COMMENTED** [review]
Good work!

**Comment by ishika2741** [comment]
Good work!

File: Problem_1.cpp (hunk 1 above)
Comment (goodguygoddy) [reply Problem_1.cpp]: Yes, space complexity would be O(amount)

File: Problem_2.cpp (hunk 2 above)
Comment (goodguygoddy) [reply Problem_2.cpp]: Yes, space complexity would be O(n)

**Pull request closed without merging**